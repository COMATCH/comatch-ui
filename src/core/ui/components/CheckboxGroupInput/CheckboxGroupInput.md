## CheckboxGroupInput

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**checkboxesPerRow** | `Enum(1, 2, 3, 4, 5)` | `1` | :x: | 
**display** | `Enum('block', 'inline')` | `'block'` | :x: | 
**id** | `String` | `null` | :x: | 
**inputError** | `String` | `''` | :x: | 
**label** | `String` | `''` | :x: | 
**name** | `String` |  | :white_check_mark: | 
**onChange** | `Function` | `noop` | :x: | 
**options** | `Union<Array[]<Shape>\|Array[]<Shape>>` |  | :white_check_mark: | 
**options<1>** | `Array[]<Shape>` |  | :x: | 
**options<1>[].label** | `Union<String\|ReactNode>` |  | :white_check_mark: | 
**options<1>[].label<1>** | `String` |  | :x: | 
**options<1>[].label<2>** | `ReactNode` |  | :x: | 
**options<1>[].value** | `Union<String\|Number>` |  | :white_check_mark: | 
**options<1>[].value<1>** | `String` |  | :x: | 
**options<1>[].value<2>** | `Number` |  | :x: | 
**options<2>** | `Array[]<Shape>` |  | :x: | 
**options<2>[].options** | `Array[]<Shape>` |  | :x: | 
**options<2>[].options[].label** | `Union<String\|ReactNode>` |  | :white_check_mark: | 
**options<2>[].options[].label<1>** | `String` |  | :x: | 
**options<2>[].options[].label<2>** | `ReactNode` |  | :x: | 
**options<2>[].options[].value** | `Union<String\|Number>` |  | :white_check_mark: | 
**options<2>[].options[].value<1>** | `String` |  | :x: | 
**options<2>[].options[].value<2>** | `Number` |  | :x: | 
**options<2>[].title** | `String` |  | :white_check_mark: | 
**required** | `Boolean` | `false` | :x: | 
**value** | `Array[]<String>` | `[]` | :x: | 

