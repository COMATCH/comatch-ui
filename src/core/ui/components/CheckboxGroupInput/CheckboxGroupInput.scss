@import '../../../styles/variables.scss';

.CheckboxGroupInput {
    margin-bottom: $co-inputs-margin-bottom;

    &:not(.inline) {
        .InputLabel {
            width: 100%;
        }
    }

    // Convenience variations where the user can choose how many
    // checkboxes the group should render per row, by setting a prop
    @for $i from 2 through 5 {
        &.per-row-#{$i} {
            display: flex;
            flex-flow: row wrap;

            .SectionHeader {
                width: 100%;

                &:not(:first-child) {
                    margin-top: 20px;
                }
            }

            .CheckboxInput {
                display: inline-block;
                margin: 0;
                flex-basis: percentage(1 / $i);
            }
        }
    }

    // states
    &.has-error {
        .CheckboxInput__label {
            color: $co-color-black;
        }

        .InputError {
            // precaution to push error to next line even if per-row is specified
            width: 100%;
        }
    }
}
