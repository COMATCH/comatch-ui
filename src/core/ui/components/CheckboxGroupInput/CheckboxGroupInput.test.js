import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { CheckboxGroupInput } from './CheckboxGroupInput';
import '../../../config/tests/setup';

describe('CheckboxGroupInput', () => {
    const options = [{ value: 'option1', label: 'Option 1' }, { value: 'option2', label: 'Option 1' }];

    const optionsWithTitle = [
        {
            title: 'Category 1',
            options: [{ value: 'option1', label: 'Option 1' }, { value: 'option2', label: 'Option 1' }],
        },
    ];
    it('should render the options correctly', () => {
        const checkBoxGroup = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} />)
            .toJSON();

        expect(checkBoxGroup).toMatchSnapshot();

        const checkBoxGroupOptionsWithTitles = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={optionsWithTitle} />)
            .toJSON();
        expect(checkBoxGroupOptionsWithTitles).toMatchSnapshot();
    });

    it('should render with display inline correctly', () => {
        const checkBoxGroup = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} display="inline" />)
            .toJSON();

        expect(checkBoxGroup).toMatchSnapshot();
    });

    it('should render with checkboxesPerRow correctly', () => {
        const checkBoxGroup = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} checkboxesPerRow={2} />)
            .toJSON();

        expect(checkBoxGroup).toMatchSnapshot();
    });

    it('should render with label/title correctly', () => {
        const checkBoxGroup = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} label="sample-label" />)
            .toJSON();

        expect(checkBoxGroup).toMatchSnapshot();

        const checkBoxGroupWithNodeTitle = renderer
            .create(
                <CheckboxGroupInput name="checkbox-sample-name" options={options} title='sample-title' />,
            )
            .toJSON();

        expect(checkBoxGroupWithNodeTitle).toMatchSnapshot();
    });

    it("should render with with label/title and required '*' correctly", () => {
        const checkBoxGroupWithNodeTitleAndRequired = renderer
            .create(
                <CheckboxGroupInput
                    name="checkbox-sample-name"
                    options={options}
                    label='sample-label'
                    required={true}
                />,
            )
            .toJSON();
        expect(checkBoxGroupWithNodeTitleAndRequired).toMatchSnapshot();
    });

    it('should render with inputError correctly', () => {
        const checkBoxGroup = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} inputError="sample-inputError" />)
            .toJSON();

        expect(checkBoxGroup).toMatchSnapshot();

        const checkBoxGroupWithNodeInputError = renderer
            .create(<CheckboxGroupInput name="checkbox-sample-name" options={options} inputError="sample-inputError" />)
            .toJSON();

        expect(checkBoxGroupWithNodeInputError).toMatchSnapshot();
    });

    it('should handle the onChange correctly', () => {
        const mockFunction = jest.fn();

        const wrapper = mount(
            <CheckboxGroupInput name="checkbox-sample-name" options={options} onChange={mockFunction} />,
        );

        wrapper
            .find('input')
            .at(0)
            .simulate('change', { target: { checked: true } });
        wrapper
            .find('input')
            .at(1)
            .simulate('change', { target: { checked: true } });

        expect(mockFunction).toHaveBeenCalledTimes(2);
    });
});
