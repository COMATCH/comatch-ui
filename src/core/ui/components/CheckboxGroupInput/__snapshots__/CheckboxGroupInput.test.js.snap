// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroupInput should render the options correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-1 sc-ifAKCX hkLpQG"
  id={null}
>
  
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render the options correctly 2`] = `
<div
  className="Input CheckboxGroupInput per-row-1 sc-ifAKCX hkLpQG"
  id={null}
>
  
  <header
    className="SectionHeader size-xs"
  >
    <span>
      <h6
        className="SectionHeader__title"
      >
        Category 1
      </h6>
    </span>
    <div
      className="SectionHeader__controls"
    />
  </header>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render with checkboxesPerRow correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-2 sc-ifAKCX kaoULA"
  id={null}
>
  
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render with display inline correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-1 inline sc-ifAKCX hJBCfa"
  id={null}
>
  
  <div
    className="Input CheckboxInput inline sc-bxivhb hkoirp"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput inline sc-bxivhb hkoirp"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render with inputError correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-1 has-error sc-ifAKCX bRgrpd"
  id={null}
>
  
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <span
    className="InputError sc-bwzfXH bZudyR"
  >
    sample-inputError
  </span>
</div>
`;

exports[`CheckboxGroupInput should render with inputError correctly 2`] = `
<div
  className="Input CheckboxGroupInput per-row-1 has-error sc-ifAKCX bRgrpd"
  id={null}
>
  
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <span
    className="InputError sc-bwzfXH bZudyR"
  >
    sample-inputError
  </span>
</div>
`;

exports[`CheckboxGroupInput should render with label/title correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-1 sc-ifAKCX hkLpQG"
  id={null}
>
  <label
    className="InputLabel sc-bdVaJa iThGRf"
    htmlFor={null}
    id={null}
  >
    sample-label
  </label>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render with label/title correctly 2`] = `
<div
  className="Input CheckboxGroupInput per-row-1 sc-ifAKCX hkLpQG"
  id={null}
>
  <label
    className="InputLabel sc-bdVaJa iThGRf"
    htmlFor={null}
    id={null}
  >
    <span>
      sample-label
    </span>
  </label>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;

exports[`CheckboxGroupInput should render with with label/title and required '*' correctly 1`] = `
<div
  className="Input CheckboxGroupInput per-row-1 sc-ifAKCX hkLpQG"
  id={null}
>
  <label
    className="InputLabel sc-bdVaJa iThGRf"
    htmlFor={null}
    id={null}
  >
    <span>
      sample-label
    </span>
    *
  </label>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option1"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  <div
    className="Input CheckboxInput block sc-bxivhb bJyOBc"
    id={null}
  >
    <label>
      <input
        checked={false}
        name="checkbox-sample-name"
        onChange={[Function]}
        type="checkbox"
        value="option2"
      />
      <div
        className="CheckboxInput__input sc-htpNat jpWaUF"
        role="checkbox"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-check fa-w-16 CheckboxInput__tick"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <span
        className="InputLabel CheckboxInput__label sc-bdVaJa eCYGDq"
        id={null}
      >
        Option 1
      </span>
    </label>
    
  </div>
  
</div>
`;
